libmscore/measure.cpp

```diff
@@ -3414,7 +3414,11 @@ qreal Measure::createEndBarLines(bool isLastMeasureInSystem)
                   }
 
             bool force = false;
-            if (repeatEnd()) {
+            if (!isLastMeasureInSystem && repeatEnd() && nextMeasure()->repeatStart()) {
```

Comment:
This is specifically the part I think we should remove.  Let's not create this in the score, let's let the scores be exactly as they are.  All this PR needs to do is handle the palette element (layout/drawing of it, what happens when you drop that barline type).  I think it's much simpler that way.